crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -746,6 +770,30 @@ impl<
                     .load_to_general_reg(&mut self.buf, src2);
                 ASM::imul_reg64_reg64_reg64(&mut self.buf, dst_reg, src1_reg, src2_reg);
             }
+            Layout::Builtin(Builtin::Int(
+                IntWidth::U64 | IntWidth::U32 | IntWidth::U16 | IntWidth::U8,
+            )) => {
+                let dst_reg = self.storage_manager.claim_general_reg(&mut self.buf, dst);
+                let src1_reg = self
+                    .storage_manager
+                    .load_to_general_reg(&mut self.buf, src1);
+                let src2_reg = self
+                    .storage_manager
+                    .load_to_general_reg(&mut self.buf, src2);
+                ASM::umul_reg64_reg64_reg64(&mut self.buf, dst_reg, src1_reg, src2_reg);
+            }
```

Comment:
here I'm running into some trouble. https://www.felixcloutier.com/x86/mul takes only one source operand. I didn't find any examples that are similar. What should I do here?

also it looks like `U64` might just be able to use the normal `imul` because the difference (the overflow) is not observable?

Reply:
`imul` is signed, so that would be noticeable as a difference.

Anyway, `mul` is an old, kinda strange, instruction that requires specific registers to functions. We would use the 64 bit version of the instruction. `Unsigned multiply (RDX:RAX ← RAX ∗ r/m64)`. So this function always affects `RAX` and `RDX`. As such, we always need to clear out `RAX` and `RDX` before running a multiply instruction, otherwise we might overwrite another variable. `ensure_reg_free` with `load_to_specified_general_reg` on the storage manager should be able to do everything you need. Free `RDX`, free `RAX`, load the param into `RAX`.